libs/libs-pomfix/lib/fix.js

```diff
@@ -0,0 +1,197 @@
+      }
+    }).get()
+
+  var artifactItem = $('plugin artifactItem')
+    .map(function (i, el) {
+      var value = function (tag) {
+        return $(el).find(tag).first().text().trim() || undefined
+      }
+
+      return {
+        group: value('groupId'),
+        artifact: value('artifactId'),
+        version: value('version'),
+        scope: value('scope'),
+        type: value('type')
+      }
+    }).get()
+
+  return [].concat(descriptor).concat(artifactItem)
+}
+
+var processFeature = function ($) {
+  var pattern = /.*mvn:([^/]+)\/([^/]+)\/([^/]+)(\/)?(.+)?/
+
+  // find the implicit dependencies in the features.xml file
+  return $('feature bundle')
+    .map(function (i, el) {
+      var bundle = $(el).text().trim().match(pattern)
+
+      return {
+        group: bundle[1],
+        artifact: bundle[2],
+        version: bundle[3],
+        type: bundle[5]
+      }
+    }).get()
+}
+
+var makeDep = function ($) {
+  var makeTag = function (tag) { return $('<' + tag + '>') }
+  return function (dep) {
+    var required = [
+      makeTag('groupId').text(dep.group),
+      makeTag('artifactId').text(dep.artifact)
+    ]
+
+    var optional = ['version', 'scope', 'classifier', 'type', 'optional']
+      .map(function (tag) {
+        if (dep[tag] !== undefined) {
+          return makeTag(tag).text(dep[tag])
+        }
+      }).filter(notUndefined)
+
+    return makeTag('dependency').append(required.concat(optional))
+  }
+}
+
+module.exports = function (load, options) {
+  var opts = options || {}
+
+  // filter out unneeded dependencies
+  var keep = function (dep) {
+    var match = function (matchers, field) {
+      if (field !== undefined) {
+        var m = matchers.map(function (pattern) {
+          return field.match(pattern)
+        }).filter(notNull)
+        if (m.length > 0) return m
+      }
+    }
+
+    var map = function (obj, dep) {
+      return Object.keys(obj || {}).reduce(function (prev, key) {
+        var m = match(obj[key], dep[key])
+        return (m !== undefined) ? prev.concat(m) : prev
+      }, [])
+    }
+
+    if (map(opts.exclude, dep).length > 0) return false
+    if (map(opts.include, dep).length > 0) return true
+
+    return false // the default policy is to exclude any dep
+  }
+
+  var find = function () {
+    var findPom = function (path) {
+      if (path.match('features.xml')) {
+        return path.replace(/src\/.+/, 'pom.xml')
+      }
+      return path
+    }
+
+    return es.through(function (entry) {
+      var deps
+      var path = entry.path
+      var $ = load(path)
+
+      if (path.match('pom.xml')) {
+        deps = processPom($)
+      } else if (path.match('features.xml')) {
+        deps = processFeature($)
+      }
+
+      if (deps !== undefined && deps.length > 0) {
+        this.emit('data', {
+          path: findPom(path),
+          deps: deps.filter(keep)
+        })
+      }
+    })
+  }
+
+  // inject deps into pom files
+  var inject = function () {
```

Comment:
Any way we can use clearer function names instead of having to use inline comments?